tests/bluetooth/host/buf/bt_buf_get_rx/src/test_suite_invalid_inputs.c

```diff
@@ -18,9 +18,9 @@
  *  Expected behaviour:
  *   - An assertion should be raised as an invalid parameter was used
  */
-void test_invalid_input_type_bt_buf_cmd(void)
+ZTEST_EXPECT_FAIL(test_bt_buf_get_rx_invalid_input, test_invalid_input_type_bt_buf_cmd);
```

Comment:
I think `ZTEST_EXPECT_FAIL` and `ZTEST_EXPECT_SKIP` are for testing the test framework itself. I'm not sure if using `ZTEST_EXPECT_FAIL` for expecting stack assertion failures in UUT is a intended use. Any comment, @yperess?

Reply:
In the [Test Framework guide lines](https://docs.zephyrproject.org/latest/develop/test/ztest.html), it's described under advanced features of the framework to be used where tests are made to be broken.

Reply:
Ok. I just wanted @yperess to confirm. I'm suspicious because I cannot think of any use case for `ZTEST_EXPECT_SKIP` other than testing the "assume"-functions themselves.

Reply:
so is `expect_assert()` still necessary, or even working?

Reply:
> so is `expect_assert()` still necessary, or even working?

Yes, it's still there and working, both approaches have the same effect.
There is only one case that `ZTEST_EXPECT_FAIL` couldn't replace the `expect_assert` so it's kept for that reason.

Reply:
I'd prefer if we kept a single approach then.

Reply:
I'm in favor of using the one supported by ZTest.
I'll have a look again at that particular case trying to find a working solution.

Reply:
What is different for that particular case :
The UT project `bt_keys_get_addr()` contains a test that needs to run at startup which checks that the keys pool list is empty and initially cleared.

The problem is that tests are sorted and ran in alphanumerical order, so the startup check isn't guaranteed to run first.

as a workaround to that, I've combined a flag with the test rule function that runs before each test case.
The test rule function skip all tests except the startup check, after that the flag is toggled and all tests are triggered again.

Now, when the test is skipped while its expected result is set to fail by `ZTEST_EXPECT_FAIL`, the framework mark that test as failed because the final result isn't as the expected result (although it was skipped)

Now, there are 2 proposals

1. Keep the current implementation which uses the `expect_assert()` in this case
2. Split the UT project into 2 projects, one will contain the startup checks only so the order won't matter and the other project will contain the rest of tests.

Please advice

Reply:
I would like us to keep using `expect_assert` everywhere, and not use `ZTEST_EXPECT_FAIL`.

Reply:
> I would like us to keep using `expect_assert` everywhere, and not use `ZTEST_EXPECT_FAIL`.

That's a bit surprising to me, let's wait for @jori-nordic.

Reply:
I agree with @alwa-nordic . I don't really like that we expect a test to fail, what does that even mean? It could fail for a different reason entirely and we wouldn't know.
In that case the testcase is trivial so it's not an issue, but if we get used to using that macro, when we make bigger test cases, we might be tempted to use it and that could hide a real issue.

Reply:
Ok, your point is totally valid.
For now, should I switch back to use `expect_assert`, or keep it and avoid using `ZTEST_EXPECT_FAIL` in the upcoming PRs.

Reply:
I don't really understand your question, but yeah you should keep using `expect_assert` and not use `ZTEST_EXPECT_FAIL`.

Reply:
Ok, that's enough for me